mod/activity/views/default/resources/activity/group.php

```diff
@@ -1,45 +1,27 @@
 <?php
 
-use Elgg\Activity\GroupRiverFilter;
+use Elgg\Exceptions\Http\EntityNotFoundException;
 
-$group_guid = elgg_extract('guid', $vars);
-
-elgg_entity_gatekeeper($group_guid, 'group');
-
-/* @var $group ElggGroup */
-$group = get_entity($group_guid);
-
-elgg_set_page_owner_guid($group->guid);
+$page_owner = elgg_get_page_owner_entity();
+if (!$page_owner instanceof ElggGroup) {
+	throw new EntityNotFoundException(elgg_echo('river:subject:invalid_subject'));
+}
 
 elgg_group_tool_gatekeeper('activity');
 
 elgg_push_breadcrumb(elgg_echo('groups'), elgg_generate_url('collection:group:group:all'));
-elgg_push_breadcrumb($group->getDisplayName(), $group->getURL());
-
-$options = [
-	'wheres' => [
-		new GroupRiverFilter($group),
-	],
-	'no_results' => elgg_echo('river:none'),
-];
+elgg_push_breadcrumb($page_owner->getDisplayName(), $page_owner->getURL());
 
+// get filter options
 $type = preg_replace('[\W]', '', get_input('type', 'all'));
```

Comment:
why not move getters to helpers view?

Reply:
input should be handled in the controller not the views